### PR TITLE
added defensive code around items that lack a name property, also add…

### LINKED ISSUE #84: Plugin type error "Cannot read property 'value' of undefined"
## Description

I fired a query to to my GraphQL server that had a schema error, a field included an argument that did not exist.

Got

```
{
  "error": {
    "errors": [
      {
        "message": "Cannot read property 'value' of undefined",
        "extensions": {
          "code": "INTERNAL_SERVER_ERROR"
        }
      }
    ]
  }
}
```

## Expected Behavior
When I remove the plugin I get:

```
{
  "error": {
    "errors": [
      {
        "message": "Unknown argument \"activityType\" on field \"User.commutes\".",
        "locations": [
          {
            "line": 8,
            "column": 16
          }
        ],
        "extensions": {
          "code": "GRAPHQL_VALIDATION_FAILED"
        }
      }
    ]
  }
}
```

## Your Environment

@newrelic/apollo-server-plugin@0.2.0 & Node 14.17.0

## Additional context

I found the line that causes the type error, `selection.name` comes through as undefined sometimes.

https://github.com/newrelic/newrelic-node-apollo-server-plugin/blob/ca870ed0bb675b77431d4a0625b237fadddaef41/lib/create-plugin.js#L357

### PR DESCRIPTION
**Pull request by bizob2828**
…ed a test to prove this failed before fix

## Proposed Release Notes
 * Fixes crash when items lack a name property(InlineFragments)

## Links
Closes #84 

## Details
You can verify the fix I made works by reverting the change and running the `tests/versioned/apollo-federation/transaction-inline-fragments.test.js`.  It will fail before the test and pass afterwards

### CODE REVIEW

File: tests/versioned/apollo-federation/federated-gateway-server-setup.js
Comment (carlo-808) [inline tests/versioned/apollo-federation/federated-gateway-server-setup.js:49]: Seems like this is a new option. We should document when with this issue: https://github.com/newrelic/newrelic-node-apollo-server-plugin/issues/88

Comment (bizob2828) [reply tests/versioned/apollo-federation/federated-gateway-server-setup.js:49]: this is just in the testing code not the plugin

Comment (carlo-808) [reply tests/versioned/apollo-federation/federated-gateway-server-setup.js:49]: Derp, vacation brain still loading ... 😄

File: tests/versioned/apollo-federation/transaction-inline-fragments.test.js
Comment (michaelgoin) [inline tests/versioned/apollo-federation/transaction-inline-fragments.test.js:50]: 👍

File: tests/versioned/apollo-federation/transaction-inline-fragments.test.js
Comment (michaelgoin) [inline tests/versioned/apollo-federation/transaction-inline-fragments.test.js:24]: An example of a breaking subgraph query could be cool in the comment here. Super minor so will approve but if you want to add may help us in the future if we forget what that looks like.

**Review by michaelgoin — DISMISSED** [review]
Looks great 👍

File: tests/versioned/apollo-federation/transaction-inline-fragments.test.js
Comment (bizob2828) [reply tests/versioned/apollo-federation/transaction-inline-fragments.test.js:24]: i can add that, don't merge this yet

**Review by michaelgoin — APPROVED** [review]
👍

**Pull request merged**
